tests/expressions/test_expressions.py

```diff
@@ -702,6 +702,10 @@ def test_or() -> None:
     # Some syntactic sugar
     assert or_ == null | nan
 
+    assert (
+        or_.model_dump_json()
+        == '{"type":"or","left":"IsNull(term=Reference(name=\'a\'))","right":"IsNaN(term=Reference(name=\'b\'))"}'
```

Comment:
This test is invalid, it should return something like:

```suggestion
        == '{"type":"or","left":{"type":"is-null","term":"a"},"right":{"type":"is-nan","term":"b"}}'
```

But the unary predicates are still WIP: https://github.com/apache/iceberg-python/pull/2598

Reply:
I think a seperate test would be best :)